output_foundry.go

```diff
@@ -333,11 +343,13 @@ func (f *FoundryOutput) checkStateChangeTransition(next ChainConstrainedOutput,
 		return fmt.Errorf("%w: maximum supply mismatch wanted %s but got %s", ErrInvalidChainStateTransition, f.MaximumSupply, nextState.MaximumSupply)
 	case f.TokenTag != nextState.TokenTag:
 		return fmt.Errorf("%w: token tag mismatch wanted %s but got %s", ErrInvalidChainStateTransition, f.TokenTag, nextState.TokenTag)
+	case f.MintedTokens.Cmp(nextState.MintedTokens) == 1:
+		return fmt.Errorf("%w: current minted supply (%s) bigger than next minted supply (%s)", ErrInvalidChainStateTransition, f.MintedTokens, nextState.MintedTokens)
+	case f.MeltedTokens.Cmp(nextState.MeltedTokens) == 1:
+		return fmt.Errorf("%w: current melted supply (%s) bigger than next melted supply (%s)", ErrInvalidChainStateTransition, f.MeltedTokens, nextState.MeltedTokens)
```

Comment:
Why isn't the foundry allowed to mint/melt in a state transition?

Reply:
These lines mean that the current state counters can not be bigger than the next state counters.

Reply:
Ok, that is reasonable.